enterprise/app/org/org_access_denied.tsx

```diff
@@ -27,15 +27,37 @@ export default class OrgAccessDeniedComponent extends React.Component<Props> {
             <div className="titles">
               <div className="title">Access denied</div>
             </div>
-            {!deniedByIpRules && <div className="details">You are not authorized to access this site.</div>}
+            {!deniedByIpRules && <div className="details">You are not authorized to access this organization.</div>}
             {deniedByIpRules && <div className="details">Access blocked by Organization IP Rules.</div>}
-            {this.props.user?.subdomainGroupID && (
+            {!deniedByIpRules && (
+              <div>
+                <FilledButton
+                  onClick={() => {
+                    window.location.href = "/join/";
+                  }}
+                  className="request-button">
+                  Request to join organization
+                </FilledButton>
+              </div>
+            )}
+            {!deniedByIpRules && (
+              <div>
+                <OutlinedButton
+                  onClick={() => {
+                    window.location.href = "/logout/";
+                  }}
+                  className="switch-button">
+                  Switch accounts
+                </OutlinedButton>
+              </div>
+            )}
+            {
               <div>
                 <FilledButton onClick={this.handleImpersonateClicked.bind(this)} className="impersonate-button">
                   Impersonate owner
```

Comment:
the impersonate button should only be shown if `this.props.user?.subdomainGroupID` is set. you might have accidentally removed that check? w/o that check the button will show up for everyone

Reply:
Good catch, I removed that to test locally what the three buttons would look like next to one another - but forgot to bring it back. Fixed.